src/template/template.js

```diff
@@ -0,0 +1,55 @@
+const path = require('path');
+const fs = require('fs-extra-promise');
+
+const requiredFiles = ['index.md', 'site.json', '_markbind/'];
+
+function Template(rootPath, templatePath) {
+  this.rootPath = rootPath;
+  this.template = templatePath;
+}
+
+function validateTemplateFromPath(templatePath) {
+  for (let i = 0; i < requiredFiles.length; i += 1) {
+    const requiredFile = requiredFiles[i];
+    const requiredFilePath = path.join(templatePath, requiredFile);
+
+    if (!fs.existsSync(requiredFilePath)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+function generateSiteWithTemplate(rootPath, templatePath) {
+  return new Promise((resolve, reject) => {
+    fs.accessAsync(rootPath)
+      .catch(() => fs.emptydirSync(rootPath))
+      .then(() => fs.copyAsync(templatePath, rootPath))
+      .then(() => {
+        resolve();
+      })
+      .catch((err) => {
+        reject(err);
+      });
```

Comment:
Does `.catch(reject)` not work for this case? I don't think we need to explicitly specify `err` unless we want to use the value (e.g. if we want to log the error message with the logger).